src/main/java/org/candlepin/subscriptions/subscription/SubscriptionSyncController.java

```diff
@@ -592,4 +594,22 @@ public List<org.candlepin.subscriptions.db.model.Subscription> findSubscriptions
 
     return result;
   }
+
+  /**
+   * This will allow any service to lookup the swatch product(s) associated with a given SKU. (This
+   * lookup will use the offering information already stored in the database) and map the
+   * `product_name` to a swatch `product_tag` via info in `tag_profile.yaml`
+   *
+   * @param sku
+   * @return OfferingProductTags
+   */
+  public OfferingProductTags findProductTags(String sku) {
+    OfferingProductTags productTags = new OfferingProductTags();
+    var productTag = offeringRepository.findProductNameBySku(sku);
+    if (productTag.isPresent()
```

Comment:
@kartikshahc Thinking this should return a 404 if the SKU isn't present in offering table.

Reply:
Not sure either. 4xx may add to splunk and this could be a valid case. Do you want me to log a warning instead of returning 404?

Reply:
I think 404 is valid in this case and on the client side we can log a warning or error if needed.  I don't think this should be an issue in splunk but paging @kahowell if that is a concern for this.

Reply:
Done